docs/mindsdb_sql/knowledge-bases.mdx

```diff
@@ -102,109 +324,124 @@ DESCRIBE KNOWLEDGE_BASE my_kb;
+| A1B      | Wireless Mouse    | Request color: black   |
+| 3XZ      | Bluetooth Speaker | Gift wrap requested    |
+| Q7P      | Laptop Stand      | Prefer aluminum finish |
++----------+-------------------+------------------------+
+```
+
+A user updated `Laptop Stand` to `Aluminum Laptop Stand`.
 
 ```sql
--- Insert data from table into a knowledge base
-INSERT INTO my_kb 
-SELECT * 
-FROM source_database.source_table
-WHERE id = 3
-USING 
-    batch_size = 200,
-    track_column = id;
++----------+-----------------------+------------------------+
+| order_id | product               | notes                  |
++----------+-----------------------+------------------------+
+| A1B      | Wireless Mouse        | Request color: black   |
+| 3XZ      | Bluetooth Speaker     | Gift wrap requested    |
+| Q7P      | Aluminum Laptop Stand | Prefer aluminum finish |
++----------+-----------------------+------------------------+
+```
+
+Here is how to propagate this change into the knowledge base.
 
--- Insert predictions (made by joining a table with a model) into a knowledge base
+```sql
 INSERT INTO my_kb
-SELECT * 
-FROM source_database.source_table
-JOIN model_name -- model created within MindsDB with CREATE MODEL
-USING 
-    batch_size = 200,
-    track_column = id,
-    threads = true;
+    SELECT order_id, product, notes
+    FROM orders
+    WHERE order_id = 'Q7P';
+```
+
+The knowledge base matches the ID value to the existing one and updates the data if required.
+
+### Optimize Data Insertion
 
--- Insert data from table into a knowledge base while skipping errors
-SELECT * 
-FROM source_database.source_table
-USING 
+In order to optimize the performance of data insertion into the knowledge base, users can set up partitions and threads to insert batches of data in parallel. This also enables tracking the progress of data insertion process including cancelling and resuming it if required.
+
+Here is an example.
+
+```sql
+INSERT INTO my_kb
+    SELECT order_id, product, notes
+    FROM orders
+USING
     batch_size = 200,
-    track_column = id,
+    track_column = order_id,
     threads = 10,
     error = 'skip';
+```
 
--- Perform prediction in threads and return data 
-INSERT INTO my_kb
-SELECT * 
-FROM source_database.source_table
-JOIN model_name -- model created within MindsDB with CREATE MODEL
-USING 
-    batch_size = 200,
-    track_column = id,
-    threads = 10;
+The parameters include the following:
+
+* `batch_size` defines the number of rows fetched per iteration to optimize data extraction from the source. It defaults to 1000.
+
+* `threads` defines threads for running partitions. Note that if the [ML task queue](/setup/custom-config#overview-of-config-parameters) is enabled, threads are used automatically. The available values for include `int` (for example, `10`), which defines the number of threads to be used, or `boolean` (for example, `true` or `false`), which defines whether to enable or disable threads.
```

Comment:
I think an example with threads included will be great since may be confusing for some users